src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,144 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exceptions.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.utils.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String SQL_CREATE = "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String SQL_FIND_BY_ID = "SELECT * FROM books WHERE id = ?";
+    private static final String SQL_FIND_ALL = "SELECT * FROM books;";
+    private static final String SQL_UPDATE = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String SQL_DELETE = "DELETE FROM books WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(SQL_CREATE,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+
+            verifyAffectedRows(statement.executeUpdate());
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add book " + book.getTitle() + "to the DB", e);
+        }
+
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Book book = null;
```

Comment:
Probably we can avoid this line of code by returning `Optional.empty` in case book was not found